### ISSUE TITLE
provide configurations for multiple PHP processes and Nginx upstream pool

### ISSUE THREAD

**Issue by jakoch** [WPN-XM/WPN-XM#323]
**it would be nice to have more PHP processes running**

- [x] add multiple configurations, 
    - so that the user can select, if he wants to run one, two or a pool of php daemons.
    - Nginx would act as a balancer and distribute requests to the pool, like described here: #202 (comment)
    - This should improve the performance and allow multiple requests.
- [x] Provide configuration options in the
  - [x] server control panel
  - [ ] webinterface
- This will not change a thing in regard to stability of PHP, until #322 is solved.
  - [x] solve 322

**Comment by duzun** [comment]
>  Nginx would act as a balancer and distribute requests to the pool, like described here: #202 (comment) This should improve the performance.

With `php-fpm` on Linux or using `spawn-fcgi` on Windows I managed to run multiple processes of PHP on a single port/socket with success. No doubt it is much better to have several PHP processes that run in parallel. And with standard `fastcgi_pass 127.0.0.1:9056;` nginx easily distributes multiple requests to PHP in parallel.

I'm not sure it'll improve performance when listening to more ports like described here [#202](https://github.com/WPN-XM/WPN-XM/issues/202#issuecomment-45089295) compared to `php-fpm` or `spawn-fcgi` on a single port. 
Does `upstream ... server ...` have any side effects compared to `fastcgi_pass`?

**Comment by jakoch** [comment]
Side-effects.. well, it's a `proxy_pass` - so it's indirect forwarding and not a direct forward anymore.
Nginx acts as a proxy sitting in between:  Nginx -> Nginx Load Balancer (Proxy) -> PHP (Pool)
And there are some interesting directives, like `health_check`, `health_passive`
http://nginx.com/resources/admin-guide/load-balancer/#health_passive.

**Comment by duzun** [comment]
Looks like there is more control over communication between Nginx and PHP with indirect forwarding.

I can think of an unpleasant side-effect tho: REMOTE_ADDR will point to Nginx server (127.0.0.1) rather than to client's IP address. Of course one could simply pass the real IP through a different header (eg X-Forwarded-For), but this implies modifying the PHP applications as well.

I've had a chance to work with a framework with encrypted index.php. Stuff like geolocation was out of my control.

**Comment by jakoch** [comment]
Update:

I've added `php-cgi-spawn\spawn.exe`. This allows to spawn multiple PHP daemons.
The spawner will be used in PHP version below v7.1.

PHP v7.1 will have this solution implemented and provide better fgci multiplexing.

**Comment by jakoch** [comment]
Implemented as part of #553  - PHP CGI Spawner 
The SCP allows to define and start multiple PHP pools at the same time.
